sdk/translation/azure-ai-translation-document/samples/async_samples/sample_translation_with_custom_model_async.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+FILE: sample_translation_with_custom_model_async.py
+
+DESCRIPTION:
+    This sample demonstrates how to create a translation job and apply custom azure translation model when doing the translation.
+
+    To set up your containers for translation and generate SAS tokens to your containers (or files)
+    with the appropriate permissions, see the README.
+
+USAGE:
+    python sample_translation_with_custom_model_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_DOCUMENT_TRANSLATION_ENDPOINT - the endpoint to your Document Translation resource.
+    2) AZURE_DOCUMENT_TRANSLATION_KEY - your Document Translation API key.
+    3) AZURE_SOURCE_CONTAINER_URL - the container SAS URL to your source container which has the documents
+        to be translated.
+    4) AZURE_TARGET_CONTAINER_URL - the container SAS URL to your target container where the translated documents
+        will be written.
+    5) AZURE_CUSTOM_MODEL_ID - the URL to your Azure custom translation model.
+"""
+
+import asyncio
+
+
+async def sample_translation_with_custom_model_async():
+    import os
+    from azure.core.credentials import AzureKeyCredential
+    from azure.ai.translation.document.aio import DocumentTranslationClient
+    from azure.ai.translation.document import (
+        DocumentTranslationInput,
+        TranslationTarget,
+    )
+
+    endpoint = os.environ["AZURE_DOCUMENT_TRANSLATION_ENDPOINT"]
+    key = os.environ["AZURE_DOCUMENT_TRANSLATION_KEY"]
+    source_container_url = os.environ["AZURE_SOURCE_CONTAINER_URL"]
+    target_container_url = os.environ["AZURE_TARGET_CONTAINER_URL"]
+    custom_model_id = os.environ["AZURE_CUSTOM_MODEL_ID"]
+
+    client = DocumentTranslationClient(endpoint, AzureKeyCredential(key))
+
+    inputs = DocumentTranslationInput(
```

Comment:
since we're only using one input, suggest using the "overloaded" begin_translation method and passing the `category_id` keyword argument.